gstools/field/base.py

```diff
@@ -26,33 +27,264 @@
+
+        self.field_data["mesh_type"] = mesh_type
+
+    def set_field_data(self, name, value):
+        """Add an attribute to this instance and add it the `field_data`
+
+        This helper method is used to create attributes for easy access
+        while using this class, but it also adds an entry to the dictionary
+        `field_data`, which is used for exporting the data.
+        """
+        setattr(self, name, value)
+        self.field_data[name] = value
+
+    def add_field(
+        self, values, name="field", is_default_field=False,
+    ):
+        """Add a field (point data) to the mesh
+
+        .. note::
+            If no field has existed before calling this method,
+            the given field will be set to the default field.
+
+        .. warning::
+            If point data with the same `name` already exists, it will be
+            overwritten.
+
+        Parameters
+        ----------
+        values : :class:`numpy.ndarray`
+            the point data, has to be the same shape as the mesh
+        name : :class:`str`, optional
+            the name of the point data
+        is_default_field : :class:`bool`, optional
+            is this the default field of the mesh?
+
+        """
+        values = np.array(values)
+        self._check_point_data(values)
+        self.point_data[name] = values
+        # set the default_field to the first field added
+        if len(self.point_data) == 1 or is_default_field:
+            self.set_field_data("default_field", name)
+
+    def __getitem__(self, key):
+        """:any:`numpy.ndarray`: The values of the field."""
+        return self.point_data[key]
+
+    def __setitem__(self, key, value):
+        self.point_data[key] = value
+
+    @property
+    def pos(self):
+        """:any:`numpy.ndarray`: The pos. on which the field is defined."""
+        return self._pos
+
+    @pos.setter
+    def pos(self, value):
+        """
+        Warning: setting new positions deletes all previously stored fields.
+        """
+        self.point_data = {self.default_field: None}
+        self._pos = value
+
+    @property
+    def field(self):
+        """:class:`numpy.ndarray`: The point data of the default field."""
+        return self.point_data[self.default_field]
+
+    @field.setter
+    def field(self, values):
+        self._check_point_data(values)
+        self.point_data[self.default_field] = values
+
+    @property
+    def value_type(self, field="field"):
+        """:any:`str`: The value type of the default field."""
+        if field in self.point_data:
+            r = value_type(self.mesh_type, self.point_data[field].shape)
+        else:
+            r = None
+        return r
+
+    @property
+    def mesh_type(self):
+        """:any:`str`: The mesh type of the fields."""
+        return self._mesh_type
+
+    @mesh_type.setter
+    def mesh_type(self, value):
+        """
+        Warning: setting a new mesh type deletes all previously stored fields.
```

Comment:
There are a few places where you delete all previously stored fields, etc. I'd recommend making one method that will properly clear the Mesh object

Reply:
What do you think, should "global" data i.e. `field_data` be deleted, if for example new `pos` are set?
I'm really unsure about that.